src/main/java/com/tasker/server/boards/rest/dao/BoardDaoImpl.java

```diff
@@ -19,4 +25,35 @@ public void create(Board board) {
         jdbcTemplate.update("INSERT INTO tasker.boards(id, title, created_at, description) VALUES (?, ?, ?, ?)",
             board.getId(), board.getTitle(), board.getCreatedAt(), board.getDescription());
     }
+
+    @Transactional
+    @Override
+    public List<Board> fetchBoards() {
+
+        String sql = "SELECT id, title, description, created_at FROM tasker.boards";
+        return  jdbcTemplate.query(sql, new BoardRowMapper());
+    }
+
+    @Override
+    public Board fetchBoard(UUID boardId) {
+
+        String sql = "SELECT id, title, description, created_at FROM tasker.boards WHERE id = ?";
+        return jdbcTemplate.queryForObject(sql, new Object[] {boardId}, new BoardRowMapper());
```

Comment:
Can't you just use `boardId` instead of `new Object[] {boardId}`?

Reply:
I don't think so, query method expects array of Objects. check out method signature 
`	@Override
	public <T> List<T> query(String sql, @Nullable Object[] args, RowMapper<T> rowMapper) throws DataAccessException {
		return result(query(sql, args, new RowMapperResultSetExtractor<>(rowMapper)));
	}`

Reply:
have you checked ?

Reply:
yep, it does not accept. it throws compiler error.